### DISCUSSION TITLE
6K notes on Google Keep. How to mass import to Standard Notes

### DISCUSSION THREAD

**Discussion by bluefootedbobby** [standardnotes/forum#2550] [General]
As above, I have over 6,000 notes (plain text) on Google Keep. Lately, Google Keep has been too slow for me and I'm giving Standard Notes a try. Is there a quick and easy way to export Google Keep notes to SN?

**Comment by stefan-yas** [comment]
Happy to have you as a user @bluefootedbobby! Sure thing - just follow [this guide](https://standardnotes.com/help/35/how-can-i-import-my-notes-from-google-keep) and your switch from Google Keep to Standard Notes should be painless :)

**Comment by bluefootedbobby** [comment]
Thanks for your response, Stefan. It's not working. I got empty outputs like this:

{
  "items": [
    {
      "uuid": "d3becde5-82c4-4508-aa6e-012347626c3f",
      "content_type": "Tag",
      "content": {
        "title": "evernote",
        "references": []
      }
    }
  ]
}

Besides, it would be a pain to upload file by file. It doesn't allow multiple file uploads. Please help.

**Comment by bluefootedbobby** [comment]
Also, I'd like to know what's the performance of Standard Notes with say 8 to 10,000 notes? Does it lag when searching or displaying?

**Comment by stefan-yas** [comment]
> It's not working. I got empty outputs like this:

Are you importing only the html files? You should ignore all the json files and only import html.

> Also, I'd like to know what's the performance of Standard Notes with say 8 to 10,000 notes? Does it lag when searching or displaying?

We haven't got any reports on performance issues when searching or displaying a large amount of notes. You might notice some lag when loading the notes first time on a new device, but you should be good after.

**Comment by bluefootedbobby** [comment]
Yes, attempted to import Google Keep .html files with https://standardnotes.com/tools Google Keep → Standard Notes converter but I got this:

{
  "items": []
}

Please see attached Google Keep .html files (intentionally renamed to .html.txt because GitHub doesn't allow .html)

[2022-01-13T10_55_41.116+08_00.html.txt](https://github.com/standardnotes/forum/files/7999338/2022-01-13T10_55_41.116%2B08_00.html.txt)
[2022-01-14T16_46_55.132+08_00.html.txt](https://github.com/standardnotes/forum/files/7999339/2022-01-14T16_46_55.132%2B08_00.html.txt)
.

**Comment by stefan-yas** [comment]
@bluefootedbobby I've recorded a video of the process which you can see below. Try to follow these exact steps and let me know of the outcome.

https://user-images.githubusercontent.com/52251483/152919899-384f8738-8a1b-4981-8757-5d5fac5cab2a.mp4

**Comment by bluefootedbobby** [comment]
Hello Stefan,

Thank you for your response again.

I have followed your instructions to the T. However, I'm still not able to import my Google Keep notes.

Please see my recorded video: http://172.104.164.35/upload20220208.mp4


Keith

________________________________

**Comment by stefan-yas** [comment]
I appreciate the recording, @bluefootedbobby! Can I ask you to try keeping the formatting in the Google Keep -> SN converter by unchecking this box and then seeing if that makes a difference?

![image](https://user-images.githubusercontent.com/52251483/153319780-8e3efc31-1a7f-41c2-828e-e4ebbfd4ae65.png)

**Comment by bluefootedbobby** [comment]
Hello Stefan,

Thanks for your help. It's not working too. keep-to-sn.json shows

{
  "items": []
}

________________________________

**Comment by atmoio** [comment]
For the record the performance with 10,000 notes on mobile would probably not be to your liking currently, since all notes need to be decrypted on launch every time. This might take several seconds on mobile. Maybe up to 10s with that many notes, depending on your device. On web/desktop this isn't a problem.

How big is your total Google Keep data size? If it's larger than say 10mb and mobile usage is important and you don't have a super specced out mobile device, I'd say you might have a rough transition.

**Comment by bluefootedbobby** [comment]
My total Google Keep backup (in zip format as provided by Google Takeout) = 42.8MB.
After unzipping, the total size of all the .json files (6,072 files) = 3.08MB, whereas the total size of all the .html files (6,072 files) = 75.8MB.

Will this pose an issue in terms of launch time and search time?

I really need an app that can fire and take a note very quickly, much like a cowboy's draw of a gun. I'll be dead if I have to wait 10 or 20 seconds.

I would also need searching to be real fast — Google-like fast that is.

At the moment, I have no problem searching on Google Keep with 6,000+ notes. The problem is the "home" screen of Google Keep which takes up some 15 seconds to load, i.e. I have to wait 15 seconds before the home screen loads in order to hit the (+) button to take a note.


Keith

________________________________

**Comment by atmoio** [comment]
Hmm 75mb is pretty big. I'd try an incremental approach to importing and see how it works out for you. Desktop/web shouldn't be a problem but mobile is the question.

The converter should now be fixed for your issue, please give it another try.

**Comment by bluefootedbobby** [comment]
Did a test with 4 random .html files from Google Keep.

Is there a way to keep these notes untitled (as in, a totally blank title). I dislike the auto renaming of these notes to "Imported note 1 from Google Keep", "Imported note 2 from Google Keep"...


Thanks,
Keith

________________________________

**Comment by abdulla771** [comment]
Hi, decided to move to Standard Notes from Google Keep, however struggling to export my Keep notes to Standard Notes. I've selected the HTML files from my Google Keep export folder, tried to import them, however it freezes around 119 notes. Is there a better way? Rather than doing it manually?
